File: packages/firebase_database/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabaseException.java
package io.flutter.plugins.firebase.database;

import java.util.HashMap;
import java.util.Map;

public class FirebaseDatabaseException extends Exception {
  private final String code;
  private final String message;
  private final Map<String, Object> additionalData;

  public FirebaseDatabaseException(String code, String message, Map<String, Object> additionalData) {
    super(message, null);

    this.code = code;
    this.message = message;
    this.additionalData = additionalData;
  }
}

File: packages/firebase_database/firebase_database_platform_interface/lib/src/method_channel/firebase_database_exception.dart
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/services.dart';

class FirebaseDatabaseException extends FirebaseException implements Exception {
  FirebaseDatabaseException({
    String? code,
    String? message,
    StackTrace? stackTrace,
  }) : super(
          plugin: 'firebase_database',
          code: code ?? 'unknown',
          message: message ?? 'An unknown error occured',
          stackTrace: stackTrace,
        );

  static FirebaseDatabaseException fromPlatformException(
    PlatformException exception,
  ) {
    return FirebaseDatabaseException(
      code: exception.code,
      message: exception.message,
    );
  }
}
